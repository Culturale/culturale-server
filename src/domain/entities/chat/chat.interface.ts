import type { IMessage } from '~/domain/entities/message/message.interface';
import type { MongoId } from '~/types/types';
export interface IChat {
  id: MongoId;
  messages?: IMessage[];
}
