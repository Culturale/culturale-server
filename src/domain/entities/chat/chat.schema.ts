import { Schema, model } from "mongoose";

import type { IChat } from "./chat.interface";

const ChatSchema = new Schema({
  messages: [{ required: false, type: Schema.Types.ObjectId }], //COLLECTION OF MESSAGE
});

const ChatModel = model<IChat>("Chat", ChatSchema);

export { ChatModel };
