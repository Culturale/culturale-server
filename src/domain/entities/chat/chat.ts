import type { IMessage } from '~/domain/entities/message/message.interface';
import type { MongoId } from '~/types/types';

import type { IChat } from './chat.interface';

export class Chat implements IChat {
  public id: MongoId;
  public messages?: IMessage[];

  constructor(id: MongoId, messages: IMessage[]) {
    this.id = id;
    this.messages = messages;
  }
}
