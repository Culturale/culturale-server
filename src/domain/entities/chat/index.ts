export * from './chat';
export * from './chat.interface';
export * from './chat.schema';
