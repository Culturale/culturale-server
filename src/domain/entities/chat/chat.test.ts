import { Types } from 'mongoose';

import type { IMessage } from '~/domain/entities/message/message.interface';

import { Chat } from './chat';
import type { IChat } from './chat.interface';

describe('Chat Entity', function () {
  let instance: IChat;

  beforeEach(function () {
    const message: IMessage[] = [];
    instance = new Chat(
      new Types.ObjectId('6423f909d626b03e800a9858'),
      message
    );
  });

  it('can be created', function () {
    expect(instance).toMatchSnapshot();
  });
});
