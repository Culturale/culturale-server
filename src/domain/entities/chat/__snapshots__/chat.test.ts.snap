// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Entity can be created 1`] = `
Chat {
  "id": "6423f909d626b03e800a9858",
  "messages": [],
}
`;
